home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,99 @@
+import requests
+from datetime import datetime
+
+from django.core.management import BaseCommand
+
+from home.models import Course, Professor, ProfessorCourse
+from home.utils import Semester
+
+class Command(BaseCommand):
+    def __init__(self):
+        super().__init__()
+        self.total_num_new_courses = 0
+        self.total_num_new_professors = 0
+
+    def add_arguments(self, parser):
+        parser.add_argument("semesters", nargs='+')
+
+    def handle(self, *args, **options):
+        t_start = datetime.now()
+        semesters = [Semester(s) for s in options['semesters']]
+        print(f"Inputted Semesters: {', '.join(s.name() for s in semesters)}")
+
+        for semester in semesters:
+            kwargs = {"semester": semester, "per_page": 100, "page": 1}
+            course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+            if "error_code" in course_data[0].keys():
+                print(f"umd.io doesn't have data for {semester.name()}!")
+                continue
+
+            print(f"Working on courses for {semester.name()}...")
+
+            while course_data:
+                for umdio_course in course_data:
+                    course = Course.unfiltered.filter(name=umdio_course['course_id']).first()
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
+                            title=umdio_course['name'],
+                            credits=umdio_course['credits'],
+                            description=umdio_course["description"]
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    self._professors(course, semester)
+                    print(course)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        # if no professors were found for `course` during `semester`
+        if isinstance(umdio_professors, dict) and 'error_code' in umdio_professors.keys():
+            return
+
+        for umdio_professor in umdio_professors:
+            if "instructor" in umdio_professor['name'].lower():
```

Comment:
is this guarding against `Instructor: TBA` entries? If so, I would compare to that string exactly. I could imagine some legitimate professor with `instructor` contained their name (a la scunthorpe).

Reply:
Yes, I'm guarding against `Instructor: TBA` but my reasoning for doing this was because the format of that string isn't consistent across all datasets. These were some of the variations I saw: "Instructor: TBA", "instructor: tba", "Instructor: tba", "instructor: TBA". I also saw some of those variations without a space between the colon and "tba", for example "Instructor:TBA". 

We could use regex or hard code all the cases I saw but the regex would be simple so that might be the best solution. Let me know your thoughts.

Reply:
I'd go with a regex, something like `instructor:\s*tba` should be good enough.